js/compat/customize-snapshots.js

```diff
@@ -0,0 +1,465 @@
+				snapshot.snapshotExpandButton.toggle( exist );
+			} );
+
+			api.state( 'snapshot-saved' ).bind( function( saved ) {
+				snapshot.snapshotButton.prop( 'disabled', saved );
+			} );
+
+			api.state( 'saved' ).bind( function( saved ) {
+				if ( saved ) {
+					snapshot.snapshotButton.prop( 'disabled', true );
+				}
+			} );
+			api.bind( 'change', function() {
+				snapshot.snapshotButton.prop( 'disabled', false );
+			} );
+
+			api.state( 'snapshot-exists' ).bind( function( exists ) {
+				var buttonText, permsMsg;
+				if ( exists ) {
+					buttonText = snapshot.data.i18n.updateButton;
+					permsMsg = snapshot.data.i18n.permsMsg.update;
+				} else {
+					buttonText = snapshot.data.i18n.saveButton;
+					permsMsg = snapshot.data.i18n.permsMsg.save;
+				}
+
+				snapshot.snapshotButton.text( buttonText );
+				if ( ! snapshot.data.currentUserCanPublish ) {
+					snapshot.snapshotButton.attr( 'title', permsMsg );
+				}
+			} );
+
+			snapshot.dirtySnapshotPostSetting.bind( function( dirty ) {
+				if ( dirty ) {
+					snapshot.snapshotButton.prop( 'disabled', false );
+				} else {
+					snapshot.snapshotButton.prop( 'disabled', ! snapshot.data.dirty );
+				}
+				snapshot.updateButtonText();
+			} );
+			snapshot.dirtyScheduleDate.bind( function( dirty ) {
+				var date;
+				if ( dirty ) {
+					date = snapshot.getDateFromInputs();
+					if ( ! date || ! snapshot.data.currentUserCanPublish ) {
+						return;
+					}
+					snapshot.snapshotButton.text( snapshot.data.i18n.scheduleButton );
+				} else {
+					snapshot.updateButtonText();
+				}
+			} );
+
+			// Submit for review button.
+			if ( ! snapshot.data.currentUserCanPublish ) {
+				publishButton.hide();
+				submitButton = wp.template( 'snapshot-submit' );
+				submitButton = $( $.trim( submitButton( {
+					buttonText: snapshot.data.i18n.submit
+				} ) ) );
+				submitButton.prop( 'disabled', ! api.state( 'snapshot-exists' ).get() );
+				submitButton.insertBefore( snapshot.snapshotButton );
+				api.state( 'snapshot-submitted' ).bind( function( submitted ) {
+					submitButton.prop( 'disabled', submitted );
+				} );
+			}
+
+			header.addClass( 'button-added' );
+		},
+
+		/**
+		 * Silently update the saved state to be true without triggering the
+		 * changed event so that the AYS beforeunload dialog won't appear
+		 * if no settings have been changed after saving a snapshot. Note
+		 * that it would be better if jQuery's callbacks allowed them to
+		 * disabled and then re-enabled later, for example:
+		 *   wp.customize.state.topics.change.disable();
+		 *   wp.customize.state( 'saved' ).set( true );
+		 *   wp.customize.state.topics.change.enable();
+		 * But unfortunately there is no such enable method.
+		 *
+		 * @return {void}
+		 */
+		resetSavedStateQuietly: function resetSavedStateQuietly() {
+			api.state( 'saved' )._value = true;
+		},
+
+		/**
+		 * Hides the future date notification used for 4.7.
```

Comment:
I am using `snapshot.statusButton.value.get()`  in `api.Snapshots` which is not available in `< 4.7` so wanted to override the method `toggleDateNotification`. But updated it to 

```
		/**
		 * Toggles the date notification.
		 *
		 * @return {void}.
		 */
		toggleDateNotification: function showDateNotification() {
			var snapshot = this;
			if ( ! _.isEmpty( snapshot.dateNotification ) ) {
				snapshot.dateNotification.toggle( ! snapshot.isFutureDate() );
			}
		}
``` 
now.